addons/l10n_id_efaktur/models/account_move.py

```diff
@@ -162,6 +162,7 @@ def _generate_efaktur_invoice(self, delimiter):
             elif (not move.partner_id.vat or len(move.partner_id.vat) < 12) and move.partner_id.l10n_id_nik:
                 invoice_npwp = move.partner_id.l10n_id_nik
             invoice_npwp = invoice_npwp.replace('.', '').replace('-', '')
+            shipping_address = hasattr(move, 'partner_shipping_id') and move.partner_shipping_id and move.partner_shipping_id.contact_address.replace('\n', '')
```

Comment:
don't be lazzy and don't use a field in a module where it's not defined.

```suggestion
            shipping_address = move._get_shipping_address_efacktur() 
```
with `_get_shipping_address_efacktur() ` returning False by default and overriden whith the module adding partner_shipping_id field

also as @jco-odoo said, we moved that field in account at some point (saas-15.2?), so please check the version where your patch would be different and please set up the forward-port bot accordingly.

thanks

Reply:
@qdp-odoo I see your point but l10n_id has no connection with sales (unlike `l10n_in`) so this would mean adding `_get_shipping_address_efacktur` in base `sale` and `account` or adding a bridge module, both solutions does not seems better than using `hasattr`

Anyway I've found a different solution using the already implemented `_get_invoice_delivery_partner_id`, so that:
* if we have the shipping partner
  * shipping partner has `l10n_id_tax_address` -> we use ``l10n_id_tax_address`
  * shipping partner has no `l10n_id_tax_address` -> we take the address of the shipping partner
* if we have no shipping partner it will default to the `move.partner_id` so it will act as before the patch